lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -59,80 +67,227 @@ describe('Lighthouse Viewer', function() {
+        strategy: interceptedUrl.searchParams.get('strategy'),
+        locale: interceptedUrl.searchParams.get('locale'),
+        utm_source: interceptedUrl.searchParams.get('utm_source'),
+      };
+      expect(params).toEqual({
+        key: 'AIzaSyAjcDRNN9CX9dCazhqI4lGR7yyQbkd_oYE',
+        url: 'https://www.example.com',
+        // Order in the api call is important to PSI!
+        category: [
+          'performance',
+          'accessibility',
+          'seo',
+          'best-practices',
+          'pwa',
+        ],
+        strategy: 'mobile',
+        // These values aren't set by default.
+        locale: null,
+        utm_source: null,
+      });
+
+      // Confirm that all default categories are used.
+      const defaultCategories = Config.getCategories(defaultConfig).map(c => c.id).sort();
+      expect(interceptedUrl.searchParams.getAll('category').sort()).toEqual(defaultCategories);
+
+      // No errors.
+      assert.deepStrictEqual(pageErrors, []);
 
+      // All categories.
+      const categoryElementIds = await getCategoryElementsIds();
       assert.deepStrictEqual(
-        elementIds.sort(),
-        expected.sort(),
-        `${category} does not have the identical audits`
+        categoryElementIds.sort(),
+        lighthouseCategories.sort(),
+        `all categories not found`
       );
-    }
-  });
 
-  it('should contain a filmstrip', async () => {
-    const filmstrip = await viewerPage.$('.lh-filmstrip');
+      // Should not clear the query string.
+      expect(await viewerPage.url()).toEqual(url);
+    });
 
-    assert.ok(!!filmstrip, `filmstrip is not available`);
-  });
+    it('should call out to PSI with specified categories', async () => {
+      psiResponse = goodPsiResponse;
+
+      const url = `${viewerUrl}?psiurl=https://www.example.com&category=seo&category=pwa`;
+      await viewerPage.goto(url);
+
+      // Wait for report to render.call out to PSI with specified categories
+      await viewerPage.waitForSelector('.lh-columns');
+
+      const interceptedUrl = new URL(interceptedRequest.url());
+      expect(interceptedUrl.origin + interceptedUrl.pathname)
+        .toEqual('https://www.googleapis.com/pagespeedonline/v5/runPagespeed');
 
-  it('should not have any unexpected audit errors', async () => {
-    function getErrors(elems, selectors) {
-      return elems.map(el => {
-        const audit = el.closest(selectors.audits);
-        const auditTitle = audit && audit.querySelector(selectors.titles);
-        return {
-          explanation: el.textContent,
-          title: auditTitle ? auditTitle.textContent : 'Audit title unvailable',
-        };
+      const params = {
+        url: interceptedUrl.searchParams.get('url'),
+        category: interceptedUrl.searchParams.getAll('category'),
+      };
+      expect(params).toEqual({
+        url: 'https://www.example.com',
+        category: [
+          'seo',
+          'pwa',
+        ],
```

Comment:
maybe test `locale` and/or `utm_source` in this test as well?

Reply:
done